src/ffi/cpython/abstract_.rs

```diff
@@ -0,0 +1,304 @@
+        where_: *const c_char,
+    ) -> *mut PyObject;
+
+    #[cfg(all(Py_3_8, not(PyPy)))]
+    pub fn _PyObject_MakeTpCall(
+        tstate: *mut PyThreadState,
+        callable: *mut PyObject,
+        args: *const *mut PyObject,
+        nargs: Py_ssize_t,
+        keywords: *mut PyObject,
+    ) -> *mut PyObject;
+}
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+const PY_VECTORCALL_ARGUMENTS_OFFSET: Py_ssize_t =
+    1 << (8 * std::mem::size_of::<Py_ssize_t>() as Py_ssize_t - 1);
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+#[inline(always)]
+pub unsafe fn PyVectorcall_NARGS(n: usize) -> Py_ssize_t {
+    assert!(n <= (Py_ssize_t::MAX as usize));
+    (n as Py_ssize_t) & !PY_VECTORCALL_ARGUMENTS_OFFSET
+}
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+#[inline(always)]
+pub unsafe fn PyVectorcall_Function(callable: *mut PyObject) -> Option<vectorcallfunc> {
+    assert!(!callable.is_null());
+    let tp = Py_TYPE(callable);
+    if PyType_HasFeature(tp, Py_TPFLAGS_HAVE_VECTORCALL) == 0 {
+        return None;
+    }
+    assert!(PyCallable_Check(callable) > 0);
+    let offset = (*tp).tp_vectorcall_offset;
+    assert!(offset > 0);
+    let ptr = (callable as *const c_char).offset(offset) as *const Option<vectorcallfunc>;
+    *ptr
+}
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+#[inline(always)]
+pub unsafe fn _PyObject_VectorcallTstate(
+    tstate: *mut PyThreadState,
+    callable: *mut PyObject,
+    args: *const *mut PyObject,
+    nargsf: usize,
+    kwnames: *mut PyObject,
+) -> *mut PyObject {
+    assert!(kwnames.is_null() || PyTuple_Check(kwnames) > 0);
+    assert!(!args.is_null() || PyVectorcall_NARGS(nargsf) == 0);
+
+    match PyVectorcall_Function(callable) {
+        None => {
+            let nargs = PyVectorcall_NARGS(nargsf);
+            _PyObject_MakeTpCall(tstate, callable, args, nargs, kwnames)
+        }
+        Some(func) => {
+            let res = func(callable, args, nargsf, kwnames);
+            _Py_CheckFunctionResult(tstate, callable, res, std::ptr::null_mut())
+        }
+    }
+}
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+#[inline(always)]
+pub unsafe fn PyObject_Vectorcall(
+    callable: *mut PyObject,
+    args: *const *mut PyObject,
+    nargsf: usize,
+    kwnames: *mut PyObject,
+) -> *mut PyObject {
+    _PyObject_VectorcallTstate(PyThreadState_GET(), callable, args, nargsf, kwnames)
+}
+
+extern "C" {
+    #[cfg(all(Py_3_8, not(PyPy)))]
+    #[cfg_attr(not(Py_3_9), link_name = "_PyObject_VectorcallDict")]
+    pub fn PyObject_VectorcallDict(
+        callable: *mut PyObject,
+        args: *const *mut PyObject,
+        nargsf: usize,
+        kwargs: *mut PyObject,
+    ) -> *mut PyObject;
+
+    #[cfg(all(Py_3_8, not(PyPy)))]
+    #[cfg_attr(not(Py_3_9), link_name = "_PyVectorcall_Call")]
+    pub fn PyVectorcall_Call(
+        callable: *mut PyObject,
+        tuple: *mut PyObject,
+        dict: *mut PyObject,
+    ) -> *mut PyObject;
+}
+
+#[cfg(all(Py_3_8, not(PyPy)))]
+#[inline(always)]
+pub unsafe fn _PyObject_FastCallTstate(
```

Comment:
Where is this function? (I'm looking at https://github.com/python/cpython/blob/v3.8.6/Include/cpython/abstract.h ).

Reply:
I'm looking at the same file but the version in `master` (so `3.10-dev`). I figured it was best for these inline functions to be as up-to-date as possible.